index.js

```diff
@@ -1,5 +1,8 @@
-const Dataset = require('./dataset')
+const rdf = require('@rdfjs/data-model')
+const Dataset = require('./Dataset')
 
-module.exports = function datasetFactory (quads = [], factory) {
-  return new Dataset(quads, factory)
+function dataset (quads) {
+  return new Dataset(quads)
 }
+
+module.exports = Object.assign({ dataset }, rdf)
```

Comment:
Could this change be extracted @vhf and a new release tagged even if the rest isn't? Refs https://gitter.im/rdfjs/public?at=5df8f727e47fb31eb7cadfc0.

Reply:
@thewilkybarkid yep, let me see what I can do!

Reply:
🎉 Would you accept TypeScript definitions here too? I'll open at DefinitelyTyped otherwise afterwards.

Reply:
TypeScript definitions would be great! 😊

Reply:
👍 

(I can extract the change if you like too.)